extension/js/content_scripts/webmail/gmail/gmail-element-replacer.ts

```diff
@@ -149,13 +149,21 @@ export class GmailElementReplacer extends WebmailElementReplacer {
   };
 
   public addSecureActionsToMessageMenu = () => {
-    $(document).on('click', 'div.aHU.hx', event => {
-      const $actionsBtn = $(event.currentTarget).find(this.sel.msgActionsBtn);
-      if ($actionsBtn.length && !$('.action_menu_message_button').length) {
-        this.addMenuButton('reply', '#r');
-        this.addMenuButton('forward', '#r3');
+    const retryInterval = 100;
+    const addButtons = () => {
+      if (!$(this.sel.msgActionsBtn).length) {
+        Catch.setHandledTimeout(addButtons, retryInterval);
+      } else {
+        $(document).on('click', 'div.aHU.hx', () => {
+          if (!$('.action_menu_message_button').length) {
+            this.addMenuButton('a_reply', '#r');
+            this.addMenuButton('a_reply_all', '#r2');
+            this.addMenuButton('a_forward', '#r3');
+          }
+        });
       }
```

Comment:
here timeout should be added inside `$(document).on('click')`, as in some cases `msgActionsMenu` element will be empty and `addMenuButton` won't find correct element:

```ts
$(document).on('click', 'div.aHU.hx', () => {
           if ($('.action_menu_message_button').length) { return }
           if ($(this.sel.msgActionsMenu).is(':empty')) {
              Catch.setHandledTimeout(addButtons, retryInterval);
           } else {
               addButtons();
           }
         });
```

Reply:
I have tried to use this but this also gives undesirable result. I'll be continuing to work on this again by tomorrow. Problem is that sometimes it completely doesn't retry, and what happened is that the event happens right after the click which result to secure actions to be added after the user tried to open the context menu for the second time.

Reply:
being dependent to $(document) and then choosing 'div.aHU.hx' is definitely not the way to get. It has some latency issue because the user may not exactly interacting with div.aHU where the button for opening context menu is located. For example, the event gets trigger right after the context menu gets closed and not when the button for opening the context menu in Gmail gets clicked.